### ISSUE TITLE
Windows Build

### ISSUE THREAD

**Issue by onurozkan** [ckeditor/ckeditor5-build-classic#26]
Hello,

Currently build scripts does not work on Windows.

Is there any plan for Windows support soon? or any hacks/tricks for now?

Thanks.

**Comment by Reinmar** [comment]
Hey!

How does it break now? It may be something simple, but we didn't test it on Windows yet so I really don't know.

As for Windows support in general, I guess it's a high time that we provide it for at least the builds. Support for the entire [development environment](https://docs.ckeditor.com/ckeditor5/latest/framework/guides/contributing/development-environment.html) is something bigger and far less developers will need it so it may wait (I hope).

**Comment by pomek** [comment]
About half of the year ago I tested our test tools on Windows and one issue which I found was invalid regexp for Windows.

I guess the same problem can occur here. When we made a config file, we assumed that Webpack will be executed on Unix. So regexps have been written for Unix paths.

If we fix the regexp for Windows, most probably the problem won't occur anymore. It's worth checking.

**Comment by onurozkan** [comment]
Tried few things.

This is the result of default build as "**npm run build**"

```
C:\Users\onur\Desktop\ckeditor5-build-classic-master>npm run build

> @ckeditor/ckeditor5-build-classic@1.0.0-alpha.1 build C:\Users\onur\Desktop\ckeditor5-build-classic-master
> npm run create-entry-file && npm run build-ckeditor


> @ckeditor/ckeditor5-build-classic@1.0.0-alpha.1 create-entry-file C:\Users\onur\Desktop\ckeditor5-build-classic-master
> ./bin/create-entry-file.js

'.' is not recognized as an internal or external command,
operable program or batch file.

npm ERR!     C:\Users\onur\AppData\Roaming\npm-cache\_logs\2017-10-12T10_13_33_333Z-debug.log
```
Later i changed create entry file script to this; "node bin/create-entry-file.js".

```
C:\Users\onur\Desktop\ckeditor5-build-classic-master>npm run build

> @ckeditor/ckeditor5-build-classic@1.0.0-alpha.1 build C:\Users\onur\Desktop\ckeditor5-build-classic-master
> npm run create-entry-file && npm run build-ckeditor


> @ckeditor/ckeditor5-build-classic@1.0.0-alpha.1 create-entry-file C:\Users\onur\Desktop\ckeditor5-build-classic-master
> node bin/create-entry-file.js

Creating the entry file...
Done.

> @ckeditor/ckeditor5-build-classic@1.0.0-alpha.1 build-ckeditor C:\Users\onur\Desktop\ckeditor5-build-classic-master
> ./bin/build-ckeditor.sh

'.' is not recognized as an internal or external command,
operable program or batch file.
```

And eventually i checked shell script, saw webpack, and changed "build-ckeditor" command to "webpack".

This is the the error i got.

```
C:\Users\onur\Desktop\ckeditor5-build-classic-master>npm run build

> @ckeditor/ckeditor5-build-classic@1.0.0-alpha.1 build C:\Users\onur\Desktop\ckeditor5-build-classic-master
> npm run create-entry-file && npm run build-ckeditor


> @ckeditor/ckeditor5-build-classic@1.0.0-alpha.1 create-entry-file C:\Users\onur\Desktop\ckeditor5-build-classic-master
> node bin/create-entry-file.js

Creating the entry file...
Done.

> @ckeditor/ckeditor5-build-classic@1.0.0-alpha.1 build-ckeditor C:\Users\onur\Desktop\ckeditor5-build-classic-master
> webpack

C:\Users\onur\Desktop\ckeditor5-build-classic-master\node_modules\@ckeditor\ckeditor5-dev-webpack-plugin\lib\replacetcalls.js:29
                                const pathToCoreTranslationPackage = result.match( /.+\/ckeditor5-core/ )[ 0 ];
                                                                                                         ^

TypeError: Cannot read property '0' of null
    at compiler.resolvers.normal.resolve (C:\Users\onur\Desktop\ckeditor5-build-classic-master\node_modules\@ckeditor\ckeditor5-dev-webpack-plugin\lib\replacetcalls.js:29:78)
    at onResolve (C:\Users\onur\Desktop\ckeditor5-build-classic-master\node_modules\enhanced-resolve\lib\Resolver.js:70:11)
    at innerCallback (C:\Users\onur\Desktop\ckeditor5-build-classic-master\node_modules\enhanced-resolve\lib\Resolver.js:143:22)
    at callbackWrapper (C:\Users\onur\Desktop\ckeditor5-build-classic-master\node_modules\enhanced-resolve\lib\createInnerCallback.js:10:21)
    at next (C:\Users\onur\Desktop\ckeditor5-build-classic-master\node_modules\tapable\lib\Tapable.js:249:35)
    at C:\Users\onur\Desktop\ckeditor5-build-classic-master\node_modules\enhanced-resolve\lib\UnsafeCachePlugin.js:39:22
    at callbackWrapper (C:\Users\onur\Desktop\ckeditor5-build-classic-master\node_modules\enhanced-resolve\lib\createInnerCallback.js:10:21)
    at innerCallback (C:\Users\onur\Desktop\ckeditor5-build-classic-master\node_modules\enhanced-resolve\lib\Resolver.js:143:22)
    at callbackWrapper (C:\Users\onur\Desktop\ckeditor5-build-classic-master\node_modules\enhanced-resolve\lib\createInnerCallback.js:10:21)
    at next (C:\Users\onur\Desktop\ckeditor5-build-classic-master\node_modules\tapable\lib\Tapable.js:249:35)
    at innerCallback (C:\Users\onur\Desktop\ckeditor5-build-classic-master\node_modules\enhanced-resolve\lib\Resolver.js:143:22)
    at callbackWrapper (C:\Users\onur\Desktop\ckeditor5-build-classic-master\node_modules\enhanced-resolve\lib\createInnerCallback.js:10:21)
    at next (C:\Users\onur\Desktop\ckeditor5-build-classic-master\node_modules\tapable\lib\Tapable.js:249:35)
    at resolver.doResolve.createInnerCallback (C:\Users\onur\Desktop\ckeditor5-build-classic-master\node_modules\enhanced-resolve\lib\DescriptionFilePlugin.js:41:24)
    at callbackWrapper (C:\Users\onur\Desktop\ckeditor5-build-classic-master\node_modules\enhanced-resolve\lib\createInnerCallback.js:10:21)
    at afterInnerCallback (C:\Users\onur\Desktop\ckeditor5-build-classic-master\node_modules\enhanced-resolve\lib\Resolver.js:165:22)
    at callbackWrapper (C:\Users\onur\Desktop\ckeditor5-build-classic-master\node_modules\enhanced-resolve\lib\createInnerCallback.js:10:21)
    at next (C:\Users\onur\Desktop\ckeditor5-build-classic-master\node_modules\tapable\lib\Tapable.js:249:35)
    at C:\Users\onur\Desktop\ckeditor5-build-classic-master\node_modules\enhanced-resolve\lib\ModuleKindPlugin.js:20:36
    at callbackWrapper (C:\Users\onur\Desktop\ckeditor5-build-classic-master\node_modules\enhanced-resolve\lib\createInnerCallback.js:10:21)
    at innerCallback (C:\Users\onur\Desktop\ckeditor5-build-classic-master\node_modules\enhanced-resolve\lib\Resolver.js:143:22)
    at callbackWrapper (C:\Users\onur\Desktop\ckeditor5-build-classic-master\node_modules\enhanced-resolve\lib\createInnerCallback.js:10:21)
    at next (C:\Users\onur\Desktop\ckeditor5-build-classic-master\node_modules\tapable\lib\Tapable.js:249:35)
    at innerCallback (C:\Users\onur\Desktop\ckeditor5-build-classic-master\node_modules\enhanced-resolve\lib\Resolver.js:143:22)
    at callbackWrapper (C:\Users\onur\Desktop\ckeditor5-build-classic-master\node_modules\enhanced-resolve\lib\createInnerCallback.js:10:21)
    at next (C:\Users\onur\Desktop\ckeditor5-build-classic-master\node_modules\tapable\lib\Tapable.js:249:35)
    at C:\Users\onur\Desktop\ckeditor5-build-classic-master\node_modules\enhanced-resolve\lib\forEachBail.js:29:14
    at callbackWrapper (C:\Users\onur\Desktop\ckeditor5-build-classic-master\node_modules\enhanced-resolve\lib\createInnerCallback.js:10:21)
    at innerCallback (C:\Users\onur\Desktop\ckeditor5-build-classic-master\node_modules\enhanced-resolve\lib\Resolver.js:143:22)
    at callbackWrapper (C:\Users\onur\Desktop\ckeditor5-build-classic-master\node_modules\enhanced-resolve\lib\createInnerCallback.js:10:21)
```

I think we are now in same page with @pomek

**Comment by onurozkan** [comment]
We can easily skip first 2 problems just by changing these lines. But cgywin required for shell script. Tried on Mac and Windows, worked well.

```
"create-entry-file": "node bin/create-entry-file.js",
"build-ckeditor": "sh bin/build-ckeditor.sh",
```

So only problem probably regexp.

**Comment by pomek** [comment]
> ```
> const pathToCoreTranslationPackage = result.match( /.+\/ckeditor5-core/ )[ 0 ];
> TypeError: Cannot read property '0' of null
> ```

Looks like the problem with regexps. It should be an easy-to-solve thing. Instead of `\/` you could use `\\\\`.

**Comment by pomek** [comment]
And we could detect whether use Unix / Windows separator path in regexp.

@Reinmar, WDYT?

**Comment by onurozkan** [comment]
changed line to this, solved the problem.

```js
const pathToCoreTranslationPackage = result.match( /.+\\ckeditor5-core/ )[ 0 ];
```

**Comment by Reinmar** [comment]
@onurozkan, great! Thanks for the details. Could you make a PR to this package with your changes?

We'll handle the regexp in https://github.com/ckeditor/ckeditor5-dev/tree/master/packages/ckeditor5-dev-webpack-plugin and see if something's left. @pomek, could you look into it?

**Comment by pomek** [comment]
Yea. `\\\\` is ok in case if you create a regexp using the constructor (`new RegExp( string )`).

**Comment by pomek** [comment]
Part of the ticket related to Webpack plugin has been reported here – https://github.com/ckeditor/ckeditor5-dev/issues/297.

**Comment by pomek** [comment]
I applied changes from #27 to other builds repositories:

- https://github.com/ckeditor/ckeditor5-build-inline/commit/370d2a1eca0ff55fa33906e0c94eceb65b8eb246,
- https://github.com/ckeditor/ckeditor5-build-balloon/commit/3c280c333f7552b6b7814b77b42f554ff5edd48f.

**Comment by Reinmar** [comment]
Thanks for the PR!

The other issue that we need to solve: https://github.com/ckeditor/ckeditor5-dev/issues/297.

**Comment by Reinmar** [comment]
I've just released @ckeditor/ckeditor5-dev-webpack-plugin@2.0.20 – this version should fix the issue.
